app/src/test/java/echo/server/EchoServerTest.java

```diff
@@ -0,0 +1,31 @@
+package echo.server;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+
+public class EchoServerTest {
+  @Test
+  void should_CreateServerSocket_When_ServerStarts() throws IOException {
+    ServerSocket serverSocket = EchoServer.createSocket(0);
+    Assertions.assertNotNull(serverSocket, "Should create server socket when the server starts");
+  }
+
+  @Test
```

Comment:
Now that you have tests, it would be great to update the README. There are some assumptions in the current README:
- This assumes the user has Intellij and JDK installations already. For a user that wants to run your server locally, it is challenging to know what IDE to use or know how/where to install any dependencies. It would be nice to have this info here as well. 
- It would also be good to add a section that says how to run the tests, either in bash or in Intellij, with both being the best!
- BONUS: It would be awesome to show the user how to call the endpoint to try out the changes. Ex: curl command.